src/meta/src/rpc/ddl_controller.rs

```diff
@@ -597,7 +596,12 @@ where
             .into_iter()
             .map(|(d, f)| Some((table_col_index_mapping.rewrite_dispatch_strategy(&d)?, f)))
             .collect::<Option<_>>()
-            .context("failed to map columns")?;
+            .ok_or_else(|| {
+                // The `rewrite` only fails if some column is dropped.
+                MetaError::invalid_parameter(
+                    "unable to drop the column due to being referenced by downstream materialized views or sinks",
```

Comment:
What about indexes?

Reply:
Theoretically, we need to drop all indexes containing the dropped column, if those indexes aren't used by other relations, but it is acceptable to throw an error for now.

Reply:
Agree that we may need to follow the behavior of postgres in the future. I've marked this as a task in the tracking issue https://github.com/risingwavelabs/risingwave/issues/6903.